crates/relayer-types/src/core/ics04_channel/channel.rs

```diff
@@ -371,13 +373,22 @@ impl FromStr for Order {
     }
 }
 
+/// The possible state variants that a channel can exhibit.
+///
+/// These are encoded with integer discriminants so that there is
+/// an easy way to compare channel states against one another. More
+/// explicitly, this is an attempt to capture the lifecycle of a
+/// channel, beginning from the `Uninitialized` state, through the
+/// `Open` state, before finally being `Closed`.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub enum State {
     Uninitialized = 0,
     Init = 1,
     TryOpen = 2,
     Open = 3,
-    Closed = 4,
+    InitUpgrade = 4,
+    TryUpgrade = 5,
+    Closed = 6,
```

Comment:
Should the `State` values match the ones in `ibc-go`, https://github.com/cosmos/ibc-go/blob/04-channel-upgrades/modules/core/04-channel/types/channel.pb.go#L31 ?